akka-actor/src/main/scala/akka/actor/FSM.scala

```diff
@@ -627,6 +628,8 @@ trait FSM[S, D] extends Actor with Listeners with ActorLogging {
     for (te ← transitionEvent) { if (te.isDefinedAt(tuple)) te(tuple) }
   }
 
+  private implicit val ec: ExecutionContext = context.dispatcher
+
```

Comment:
This is only used in `scheduleTimeout` which happens on transition when there is a state timeout defined, so safe to be the original import.